.pre-commit-config.yaml

```diff
@@ -18,6 +18,26 @@ repos:
     language: system
     files: >-
       ^setup\.py$
+  - id: check-python-packaging
+    name: Check python packaging
+    description: |
+        Performs validations of python packaging using both PEP-517 and old
+        setup.py while also validating metadata. Also verifies that the package
+        can be insalled in editable mode.
+    entry: >
+      bash -c "
+      rm -rf $VIRTUAL_ENV/dist.old $VIRTUAL_ENV/dist.new &&
+      python setup.py sdist bdist_wheel --dist-dir $VIRTUAL_ENV/dist.old &&
```

Comment:
This isn't supposed to be used directly. Use it through PEP517. Also, it's unnecessary given that it's already being tested in the CI anyway.

Reply:
(when we dropped `setup.py` shims we made an assumption that end-users will get wheels that we publish and won't build from sdist most of the time and if they do it's documented that that have to use a good version of pip)